pkg/activator/handler/concurrency_reporter.go

```diff
@@ -65,12 +78,25 @@ func (cr *ConcurrencyReporter) report(key types.NamespacedName, concurrency, req
 
 	// Send the stat to another goroutine to transmit
 	// so we can continue bucketing stats.
-	cr.statChan <- &autoscaler.StatMessage{
+	cr.statCh <- &autoscaler.StatMessage{
 		Key:  key,
 		Stat: stat,
 	}
 }
 
+func (cr *ConcurrencyReporter) reportToMetricsBackend(key types.NamespacedName, concurrency int32) {
+	ns := key.Namespace
+	revName := key.Name
+	revision, err := cr.rl.Revisions(ns).Get(revName)
+	if err != nil {
+		cr.logger.Errorw("Error while getting revision", zap.Error(err))
+		return
+	}
+	configurationName := revision.Labels[serving.ConfigurationLabelKey]
+	serviceName := revision.Labels[serving.ServiceLabelKey]
+	cr.sr.ReportRequestConcurrency(ns, serviceName, configurationName, revName, int64(concurrency))
```

Comment:
Do we want this to be async same way as report to autoscaler is? Since it seems it might involve a call to the backend?

Reply:
This only records the data in memory. The reporting part is async and controlled by another time interval.